frontend/src/HomePage/HomePage.jsx

```diff
@@ -86,6 +88,22 @@ class HomePage extends React.Component {
     this.setState({ showAppDeletionConfirmation: true, appToBeDeleted: app });
   }
 
+  cloneApp = (app) => {
```

Comment:
I think there is no parameter `app` passed to this function from the below snippet
```
  <div className="field mb-2">
    <span className="field mb-2" role="button" onClick={() => cloneApp()}>
      Clone app
    </span>
  </div>
```
I think `app` is being used from the global parameter